### ISSUE TITLE
Bug: [no-unused-vars] `/* exported */` doesn't work

### ISSUE THREAD

**Issue by dead-claudia** [typescript-eslint/typescript-eslint#5513]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.4&sourceType=module&code=PQKgBApgHgDg9gJwC4QCZhMAsAKADYRJgBmccYAvGAIxA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
/* exported */
let foo = 1
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No error

### Actual Result

```
'foo' is assigned a value but never used. 2:5 - 2:8
```

### Versions

| package | version |
| -- | -- |
| `@typescript-eslint/eslint-plugin` | `5.33.1` |
| `@typescript-eslint/parser` | `5.33.1` |
| `TypeScript` | `4.7.4` |
| `ESLint` | `8.15.0` |
| `node` | `web` |

**Comment by bradzacher** [comment]
Closing as working as intended as per comments on #5511
